frontend/simple/js/state.js

```diff
@@ -110,6 +110,28 @@ const getters = {
   groupsByName (state) {
     return _.map(_.keys(_.pickBy(state.contracts, (value, key) => value.type === 'GroupContract')), key => ({groupName: state[key].groupName, contractId: key}))
   },
+  proposals (state) {
+    let proposals = []
+    if (!state.currentGroupId) { return proposals }
+    for (let groupContractId of Object.keys(state.contracts)
+      .filter(key => state.contracts[key].type === 'GroupContract')
+    ) {
+      for (let proposal of Object.keys(state[groupContractId].proposals || {})) {
+        if (state[groupContractId].proposals[proposal].initatior !== state.loggedIn.name &&
+        !state[groupContractId].proposals[proposal].for.find(name => name === state.loggedIn.name) &&
+        !state[groupContractId].proposals[proposal].against.find(name => name === state.loggedIn.name)
+        ) {
```

Comment:
This is insanely complicated. Note to self: see if there's a way to simplify this